pkg/node/chain.go

```diff
@@ -0,0 +1,249 @@
+				" endpont using --swap-endpoint",
+			endpoint,
+		)
+		err = fmt.Errorf("could not get chain ID from endpoint %s: %w", endpoint, err)
+		return
+	}
+	chainID = cID.Int64()
+
+	return
+}
+
+// InitChequebookFactory will initialize the chequebook factory with the given
+// chain backend.
+func InitChequebookFactory(
+	log logging.Logger,
+	backend *ethclient.Client,
+	chainID int64,
+	transactionService transaction.Service,
+	factoryAddress string,
+) (
+	chequebookFactory chequebook.Factory,
+	err error,
+) {
+	var addr common.Address
+	if factoryAddress == "" {
+		var found bool
+		addr, found = chequebook.DiscoverFactoryAddress(chainID)
+		if !found {
+			return nil, errors.New("no known factory address for this network")
+		}
+		log.Infof("using default factory address for chain id %d: %x", chainID, addr)
+	} else if !common.IsHexAddress(factoryAddress) {
+		return nil, errors.New("malformed factory address")
+	} else {
+		addr = common.HexToAddress(factoryAddress)
+		log.Infof("using custom factory address: %x", factoryAddress)
+	}
+
+	chequebookFactory, err = chequebook.NewFactory(
+		backend,
+		transactionService,
+		addr,
+		chequebook.NewSimpleSwapFactoryBindingFunc,
+	)
+	if err != nil {
+		err = fmt.Errorf("failed to init chequebook factory: %w", err)
+	}
+
+	return
+}
+
+// InitChequebookService will initialize the chequebook service with the given
+// chequebook factory and chain backend.
+func InitChequebookService(
+	ctx context.Context,
+	log logging.Logger,
+	stateStore storage.StateStorer,
+	signer crypto.Signer,
+	chainID int64,
+	backend *ethclient.Client,
+	overlayEthAddress common.Address,
+	transactionService transaction.Service,
+	chequebookFactory chequebook.Factory,
+	initialDeposit string,
+) (
+	chequebookService chequebook.Service,
+	err error,
+) {
+	chequeSigner := chequebook.NewChequeSigner(signer, chainID)
+
+	isSynced, err := transaction.IsSynced(ctx, backend, maxDelay)
+	if err != nil {
+		err = fmt.Errorf("failed to sync with the Ethereum backend: %w", err)
+		return
+	}
+
+	if !isSynced {
+		log.Infof("waiting to sync with the Ethereum backend")
+		err = transaction.WaitSynced(ctx, backend, maxDelay)
+		if err != nil {
+			err = fmt.Errorf("could not wait to sync with the Ethereum backend: %w", err)
+			return
+		}
+	}
+
+	deposit, ok := new(big.Int).SetString(initialDeposit, 10)
+	if !ok {
+		err = fmt.Errorf("invalid initial swap deposit: %w", err)
+		return
+	}
+
+	chequebookService, err = chequebook.Init(
+		ctx,
+		chequebookFactory,
+		stateStore,
+		log,
+		deposit,
+		transactionService,
+		backend,
+		chainID,
+		overlayEthAddress,
+		chequeSigner,
+		chequebook.NewSimpleSwapBindings,
+		chequebook.NewERC20Bindings,
+	)
+	if err != nil {
+		err = fmt.Errorf("failed to init chequebook service: %w", err)
+		return
+	}
+
+	return
+}
+
+func initChequeStoreCahsout(
+	stateStore storage.StateStorer,
+	swapBackend *ethclient.Client,
```

Comment:
use `transaction.Backend` here

Reply:
Updated to use interface. Should the same apply elsewhere in the codebase?